### PR TITLE
HPCC-23473 Update roxie to support new config paradigm

### PR DESCRIPTION
**Pull request by richardkchapman**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [x] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Please review

**Comment by richardkchapman** [comment]
@ghalliday Please review

File: roxie/ccd/ccdmain.cpp [hunk 1]
```diff
@@ -537,6 +525,29 @@ void readStaticTopology()
     createStaticTopology(allRoles, traceLevel);
 }
 
+static constexpr const char * defaultJson = R"!!({
+  "version": "1.0",
+  "Roxie": {
+    "allFilesDynamic": "1",
+    "numChannels": "1",
```
Comment (ghalliday) [inline roxie/ccd/ccdmain.cpp]: minor: Should these be unquoted, and localSlave a boolean?  I know it makes no difference in the current implementation.

File: system/jlib/jptree.cpp [hunk 2]
```diff
@@ -4093,7 +4093,15 @@ class CommonReaderBase : public CInterface
     }
     inline void skipWS()
     {
-        while (isspace(nextChar)) readNext();
+        while (isspace(nextChar) || nextChar=='#')
+        {
+            if (nextChar=='#')
+            {
+                while (nextChar != '\n')
```
Comment (ghalliday) [inline system/jlib/jptree.cpp]: need to cope with # on the end of the file with no \n (and osx files that only end in \r)

**Review by ghalliday — COMMENTED** [review]
@richardkchapman a couple of small comments

**Comment by ghalliday** [comment]
@jakesmith please can you check the code in jptree that aims to treat #... as whitespace, especially coping with /r /n and a file ending in the middle of a comment.  (Should that be treated as an error?)

**Comment by richardkchapman** [comment]
I think that's actually @afishbeck code

**Comment by richardkchapman** [comment]
@ghalliday I have taken out the json changes - they are not really needed anyway, and can wait for proper yaml support.

**Pull request merged**